cpp/src/umap/simpl_set_embed/optimize_batch_kernel.cuh

```diff
@@ -325,7 +541,7 @@ __global__ void optimize_batch_kernel(
   }
   // storing gradients for positive samples back to global memory
   if (use_shared_mem) {
```

Comment:
We should be able to get rid of this now that we're doing a warp-level reduction.